components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/context/SessionContext.java

```diff
@@ -90,4 +90,30 @@ public Object getProperty(String key) {
     public SessionAuthHistory getSessionAuthHistory() {
         return sessionAuthHistory;
     }
+
+    public void setSessionAuthHistory(SessionAuthHistory sessionAuthHistory) {
+
+        this.sessionAuthHistory = sessionAuthHistory;
+    }
+
+    public Map<String, Object> getProperties() {
+
+        return properties;
+    }
+
+    public void setProperties(Map<String, Object> properties) {
+
+        this.properties = properties;
+    }
+
+    public Map<String, Map<String, AuthenticatedIdPData>> getAuthenticatedIdPsOfApp() {
+
+        return authenticatedIdPsOfApp;
+    }
+
+    public void setAuthenticatedIdPsOfApp(
+            Map<String, Map<String, AuthenticatedIdPData>> authenticatedIdPsOfApp) {
+
+        this.authenticatedIdPsOfApp = authenticatedIdPsOfApp;
+    }
```

Comment:
let's revisit the need to introduce these public methods

Reply:
We need to set and get the attributes for the optimization purpose, As this class doesn't have the getters and setters for the respective attributes. We have added that.